src/Enzo/enzo_EnzoMethodPmDeposit.cpp

```diff
@@ -165,172 +191,187 @@ void EnzoMethodPmDeposit::compute ( Block * block) throw()
 
       for (int ib=0; ib<particle.num_batches(it); ib++) {
 
-        const int np = particle.num_particles(it,ib);
+        int np = particle.num_particles(it,ib);
+
+        // Find the mass of each particle. If cosntant generate array of
+        // constant values in order to simply code below
+        enzo_float * pdens = NULL;
+        if (ia_m > 0){
+          pdens = (enzo_float *) particle.attribute_array( it, ia_m, ib);
+        } else {
+          pdens = new enzo_float[np];
+          std::fill_n(pdens, np, 1.0);
+        }
```

Comment:
Now pdens is used for particle mass? But is an array of 1s when the mass is constant?

Reply:
This is effectively a hack I used to deal with both potential cases for particle mass without having to write an entire separate routine for each case.... this is when either A) the particle type has a constant mass (and therefor no mass attribute), or B) the particle type does not have a constant mass and this must be obtained from the mass attribute.... using the `dens` variable and `pdens` array, this can be done such that when particle mass is constant, `dens` contains this value and `pdens` is 1's, so is effectively ignored, and when particle mass is not constant, `pdens` holds the particle masses and `dens` is 1 so is effectively ignored

Reply:
Okay I can see that. This is related to my other comments on particle masses vs densities. It's something that needs to be cleared up but I can try and do this myself.